File: Data Structures & Algorithms/max-water-container/submission-2.java
class Solution {
    public int maxArea(int[] heights) {
        int n= heights.length;
        int st=0;
        int end = n -1;
        int maxWater = Integer.MIN_VALUE;

        while(st < end){
            int len = end-st;
            int curWater = len * Math.min(heights[st],heights[end]);
            maxWater=Math.max(curWater,maxWater);

            if(heights[st] < heights[end]) st++;
            else end --;
        }

        return maxWater;
        
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cs

public class Solution {
    public int LengthOfLongestSubstring(string s) {
        var map = new Dictionary<int,int>();
        int maxLength=0;
        int st=0;
        for(int i=0; i<s.Length;i++){
         
          char c = s[i];
            if(map.ContainsKey(c) && map[c] >= st){
               st = map[c]+1;
            }
            map[c] = i;
            maxLength = Math.Max(maxLength,i-st+1);
        }

        return maxLength;
    }
}
